### PR TITLE
Parse rows as maps

### PR DESCRIPTION
**Pull request by joshuawright11**
If the presto column is a `row(...)` (aka `struct`) this parses it into a `map[string]driver.Value` where `string` is the key of the struct field & `driver.Value` is the presto value converted to go.

e.g. consider table `my_table` with `string_field: varchar` `row_field: row(some_string: varchar, some_int: bigint)`

```go
rows, _ = prestodb.Query("SELECT string_field row_field FROM my_table")
for rows.Next() {
    var s string
    var m map[string]driver.Value
    err := rows.Scan(&s, &m)
    nestedString := m["some_string"].(string)
}
```

### CODE REVIEW

**Comment by joshuawright11** [comment]
cc @adityasinghhh

File: presto/presto.go [hunk 1]
```diff
@@ -869,10 +869,70 @@ func (c *typeConverter) ConvertValue(v interface{}) (driver.Value, error) {
...
+				return nil, fmt.Errorf("there should be as many types as there are values")
+			}
+
+			m := map[string]driver.Value{}
+
+			for pos, item := range rowItems {
+				converter := newTypeConverter(item.typeName)
+				value, err := converter.ConvertValue(values[pos])
+				if err != nil {
+					fmt.Printf("Error converting value of %s: %s\n", item.name, err.Error())
```
Comment (adityasinghhh) [inline presto/presto.go]: We shouldn't use prints in production code. Should we just return from here?
```suggestion
					return nil, err
```

File: presto/presto.go [hunk 2]
```diff
@@ -869,10 +869,70 @@ func (c *typeConverter) ConvertValue(v interface{}) (driver.Value, error) {
...
+				} else {
+					m[item.name] = value
+					fmt.Printf("type passed %s\n", item.typeName)
```
Comment (adityasinghhh) [inline presto/presto.go]: Drop the print.

File: presto/presto.go [hunk 3]
```diff
@@ -869,10 +869,70 @@ func (c *typeConverter) ConvertValue(v interface{}) (driver.Value, error) {
 		}
 		return v, nil
 	default:
```
Comment (adityasinghhh) [inline presto/presto.go]: Instead of `default` should we case the exact type here?

**Review by adityasinghhh — APPROVED** [review]

**Pull request closed without merging**